Branch-SDK/src/main/java/io/branch/referral/TrackingController.java

```diff
@@ -57,6 +57,7 @@ private void onTrackingDisabled(Context context) {
         prefHelper.setAppLink(PrefHelper.NO_STRING_VALUE);
         prefHelper.setInstallReferrerParams(PrefHelper.NO_STRING_VALUE);
         prefHelper.setAppStoreReferrer(PrefHelper.NO_STRING_VALUE);
+        prefHelper.setAppStoreSource(PrefHelper.NO_STRING_VALUE);
```

Comment:
Did some more testing and found an issue, if tracking was enabled and the store was set, and then tracking disabled, we would not clear it in this function.

Because we set `NO_STRING_VALUE` here for disable tracking, `setAppStoreSource` must allow it, and must check for it at POST construction so that we don't include the field. 

We can however at least use the one condition to check for this value only @JagadeeshKaricherla-branch